pcdsdevices/lens.py

```diff
@@ -286,31 +341,294 @@ def _make_safe(self):
+    att_obj : attenuator object, optional
+    lcls_obj
+        Object that gets PVs from lcls (for energy)
+    mono_obj
+        Object that gets energy from monochromator
+    beamsize_unfocused : float, optional
+        Radial size of x-ray beam before focusing.
+    path : str
+        Path to the file that defines which lenses are being used.
+
+    Examples
+    --------
+
+    Before using the LclsStack class, provide the path of the be lens set file
+    to be used for the calculations.
+
+    If no lens sets are added in the file yet, use the
+    `be_lens_calcs.set_lens_set_to_file` function to set the lens sets:
+
+    >>> import pcdsdevices.lens as lens
+    >>> import pcdscalc.be_lens_calcs as be
+
+    >>> path = '../path/to/lens_set'
+
+    >>> sets_list = [[3, 0.0001, 1, 0.0002],
+                     [1, 0.0001, 1, 0.0003, 1, 0.0005],
+                     [2, 0.0001, 1, 0.0005]]
+    >>> be.set_lens_set_to_file(sets_list, path)
+
+    Create the LensStack() object by providing the x, y and z `prefixes`, as
+    well as all the other parameters:
+
+    >>> att = Attenuator('att', 4, name='att')
+
+    >>> be_stack = lens.LensStack(path=path, x_prefix='X:PREF',
+                                  y_prefix='Y:PREF', z_prefix='Z:PREF',
+                                  att_obj=att, z_offset=2, z_dir=0.2, E=8,
+                                  name='be_stack')
+
+    For this documentation purposes we will use the SimLensStack:
+
+    >>> sim = lens.SimLensStack(path=path, x_prefix='x', y_prefix='y',
+                                z_prefix='z', z_offset=2, z_dir=0.2, E=8,
+                                att_obj=att, name='sim')
+    FWHM at lens   : 5.000e-04
+    waist          : 3.113e-07
+    waist FWHM     : 3.666e-07
+    rayleigh_range : 1.965e-03
+    focal length   : 2.680e+00
+    size           : 2.092e-04
+    size FWHM      : 2.463e-04
+
+    You can check what current sets are in the file as follows:
+
+    >>> sim.read_lens(print_only=True)
+    [[3, 0.0001, 1, 0.0002], [1, 0.0001, 1, 0.0003, 1, 0.0005],
+    [2, 0.0001, 1, 0.0005]]
+
+    If `print_only` is `False`, `self.lens_pack` will be set to use the
+    current sets:
+
+    >>> sim.read_lens()
+    [[3, 0.0001, 1, 0.0002],
+    [1, 0.0001, 1, 0.0003, 1, 0.0005],
+    [2, 0.0001, 1, 0.0005]]
+
+    >>> sim.lens_pack
+    [[3, 0.0001, 1, 0.0002],
+    [1, 0.0001, 1, 0.0003, 1, 0.0005],
+    [2, 0.0001, 1, 0.0005]]
+
+    You can also create the lens sets using this class like so:
+
+    >>> sets_list = [[3, 0.0001, 1, 0.0002],
+                     [1, 0.0001, 1, 0.0003, 1, 0.0005],
+                     [2, 0.0001, 1, 0.0005]]
+    >>> sim.create_lens(sets_list)
+
+    For calculations, one set at the time will be used. Use the `set_lens_set`
+    method to choose what set to use. Ex. to use the second set:
+
+    >>> sim.set_lens_set(2)
+
+    >>> sim.lens_set
+    [1, 0.0001, 1, 0.0003, 1, 0.0005]
+
+    Now you can try to tweak the motors:
```

Comment:
Here is where the docstring information could be way off, because I don't know exactly how we do it - any help here would be very appreciated

Reply:
`.tweak()` seems fine, though `.umvr()` (relative motion, effectively a "tweak") could be more compact for an example here.